tutorial/creating_packages/other_types_of_packages/tool_requires_packages.rst

```diff
@@ -1,2 +1,226 @@
+    ...
+    conanfile.py (my_consumer/1.0): RUN: secure_scanner /Users/luism/workspace/examples2/tutorial/creating_packages/other_packages/tool_requires/consumer
+    ...
+    Security Scanner: The path '/Users/luism/workspace/examples2/tutorial/creating_packages/other_packages/tool_requires/consumer' is secure!
+    ...
+    MY_VAR=23
+
+
+We can see that the executable returned 0 (because our folder is secure) and it printed ``Security Scanner: The path is secure!`` message.
+It also printed the "23" value assigned to ``MY_VAR`` but, why are these automatically available?
+
+- The generators ``VirtualBuildEnv`` and ``VirtualRunEnv`` are automatically used.
+- The ``VirtualRunEnv`` is reading the ``tool-requires`` and is creating a launcher like ``conanbuildenv-release-x86_64.sh`` appending
+  all ``cpp_info.bindirs`` to the ``PATH``, all the ``cpp_info.libdirs`` to the ``LD_LIBRARY_PATH`` environment variable and
+  declaring each variable of ``self.buildenv_info``.
+- Every time conan executes the ``self.run``, by default, activates the ``conanbuild.sh`` file before calling any command.
+  The ``conanbuild.sh`` is including the ``conanbuildenv-release-x86_64.sh``, so the application is in the PATH
+  and the enviornment variable "MYVAR" has the value declared in the tool-require.
+
+
+
+Removing settings in package_id()
+---------------------------------
+
+With the previous recipe, if we call :command:`conan create` with different setting like different compiler versions, we will get
+different binary packages with a different ``package ID``. This might be convenient to, for example, keep better traceability of
+our tools. In this case, the <MISSING PAGE> compatibility.py plugin can help to locate the best matching binary in case Conan doesn't find the
+binary for our specific compiler version.
+
+But in some cases we might want to just generate a binary taking into account only the ``os``, ``arch`` or at most
+adding the ``build_type`` to know if the application is built for Debug or Release. We can add a ``package_id()`` method
+to remove them:
+
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    import os
+    from conan import ConanFile
+    from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+    from conan.tools.files import copy
+
+
+    class secure_scannerRecipe(ConanFile):
+        name = "secure_scanner"
+        version = "1.0"
+        settings = "os", "compiler", "build_type", "arch"
+        ...
+
+        def package_id(self):
+            del self.info.settings.compiler
+            del self.info.settings.build_type
+
+
+So, if we call :command:`conan create` with different ``build_type`` we will get exactly the same binary.
+
+
+.. code-block:: bash
+
+    $ conan create .
+    ...
+    Package '82339cc4d6db7990c1830d274cd12e7c91ab18a1' created
+
+    $ conan create . -s build_type=Debug
+    ...
+    Package '82339cc4d6db7990c1830d274cd12e7c91ab18a1' created
+
+We got the same binary package.
```

Comment:
```suggestion
We got the same binary ``package_id``. The second ``conan create . -s build_type=Debug`` created and overwrote (created a newer package revision) of the previous Release binary, because they have the same ``package_id`` identifier. It is typical to create only the ``Release`` one, and if for any reason managing both Debug and Release binaries is intended, then the approach would be not removing the ``del self.info.settings.build_type``
```